lib/gateway/Shard.js

```diff
@@ -2069,6 +2069,220 @@ class Shard extends EventEmitter {
+                * @prop {Guild} guild The guild where threads are being synced
+                * @prop {Array<NewsThreadChannel | PrivateThreadChannel | PublicThreadChannel | Object>} deletedThreads An array of synced threads that the current user no longer has access to. If a thread channel is uncached, it will be an object with an `id` key. No other property is guaranteed
+                * @prop {Array<NewsThreadChannel | PrivateThreadChannel | PublicThreadChannel>} activeThreads An array of synced active threads that the current user can access
+                * @prop {Array<ThreadMember>} joinedThreadsMember An array of thread member objects where the current user has been added in a synced thread channel
+                */
+                this.emit("threadListSync", guild, deletedThreads, activeThreads, joinedThreadsMember);
+                break;
+            }
+            case "THREAD_MEMBER_UPDATE": {
+                const channel = this.client.getChannel(packet.d.id);
+                if(!channel) {
+                    this.emit("debug", `Missing channel ${packet.d.id} in THREAD_MEMBER_UPDATE`);
+                    break;
+                }
+                let oldMember = null;
+                // Thanks Discord
+                packet.d.thread_id = packet.d.id;
+                let member = channel.members.get((packet.d.id = packet.d.user_id));
+                if(member) {
+                    oldMember = {
+                        flags: member.flags
+                    };
+                }
+                member = channel.members.update(packet.d, this.client);
+                /**
+                * Fired when a thread member is updated
+                * @event Client#threadMemberUpdate
+                * @prop {NewsThreadChannel | PrivateThreadChannel | PublicThreadChannel} channel The channel
+                * @prop {ThreadMember} member The updated thread member
+                * @prop {Object} oldMember The old thread member data
+                * @prop {Number} oldMember.flags User thread settings
+                */
+                this.emit("threadMemberUpdate", channel, member, oldMember);
+                break;
+            }
+            case "THREAD_MEMBERS_UPDATE": {
+                const channel = this.client.getChannel(packet.d.id);
+                if(!channel) {
+                    this.emit("debug", `Missing channel ${packet.d.id} in THREAD_MEMBERS_UPDATE`);
+                    break;
+                }
+                channel.update(packet.d);
+                let removedMembers;
+                let addedMembers;
+                if(packet.d.removed_member_ids) {
+                    removedMembers = packet.d.removed_member_ids.map((id) => channel.members.remove({id}) || {id});
+                }
+                if(packet.d.added_members) {
+                    addedMembers = packet.d.added_members.map((m) => {
+                        m.presence.id = m.presence.user.id;
```

Comment:
`m.presence` can be null sometimes.

Reply:
Discord at it again with the poor documentation